packages/frontend/apps/electron/src/main/protocol.ts

```diff
@@ -110,7 +110,8 @@ export function registerProtocol() {
     const protocol = url.protocol;
     const origin = url.origin;
 
-    const sameOrigin = origin === CLOUD_BASE_URL || protocol === 'file:';
+    const sameSite =
+      url.host === new URL(CLOUD_BASE_URL).host || protocol === 'file:';
```

Comment:
is it possible to migrate from listening 'file://' protocol to 'server_origin' and do the same file resolving like the file:// way, by which we won't need to manage cookies manually

Reply:
We can try this solution this iteration. The fix in this pr need to be released in next patch.